app/models/concerns/interval.rb

```diff
@@ -12,7 +12,10 @@ module Interval
     scope :finished, -> { where.not(finished_on: nil) }
     scope :earliest_first, -> { order(started_on: 'asc') }
     scope :latest_first, -> { order(started_on: 'desc') }
-
+    scope :started_before, ->(date) { where(started_on: ...date) }
+    scope :started_on_or_after, ->(date) { where(started_on: date..) }
+    scope :finished_before, ->(date) { where(finished_on: ...date) }
+    scope :finished_on_or_after, ->(date) { where(finished_on: date..) }
     scope :containing_period, ->(period) { where("range @> daterange(?, ?)", period.started_on, period.finished_on) }
   end
```

Comment:
would it make sense to extend this concern with this:
```
  scope :siblings_of, ->(instance) {  raise ImplementationError }

  def sibblings = self.class.sibblings_of(self)
  def predecessors = sibblings.started_before(started_on)
  def successors = sibblings.started_on_or_after(finished_on)

  def sibblings? = sibblings.exists?
  def predecessors? = predecessors.exists?
  def successors? = successors.exists?
  
```

so that each `*Period` class needs to implement:
   ```scope :siblings_of, ->(instance) { ... }```

and therefore each `*Period` instance can give quick access to their sibblings, predecessors and successors instances?

Reply:
It sounds like a good idea but I think it's worth doing in a separate PR because the sibling behaviour isn't always obvious. In a mentorship, for example, is it a mentorship from the ECT's point of view, or the mentor's? We have `mentee_siblings` there at the moment. Similarly training periods are also more complex.